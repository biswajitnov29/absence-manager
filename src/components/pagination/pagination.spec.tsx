import React from "react";
import { render, screen } from "@testing-library/react";
import SitePagination from "./pagination";

describe('<SitePagination />', () => {
    test('renders correctly', () => {
        const { container } = render(<SitePagination listSize={20} pageSize={1} />);
        expect(container).toMatchSnapshot();
    });
    test('check 20th page exist', () => {
        render(<SitePagination listSize={200} pageSize={10} />);
        expect(screen.getByText('20')).toBeVisible();
    });
});