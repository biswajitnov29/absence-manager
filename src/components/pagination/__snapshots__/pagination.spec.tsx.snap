// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SitePagination /> renders correctly 1`] = `
<div>
  <nav
    class="pagination"
  >
    <ul>
      <li
        class="disabled"
      >
        <span>
          &lt;
        </span>
      </li>
      <li
        class="active"
      >
        <span>
          1
        </span>
      </li>
      <li
        class=""
      >
        <span>
          2
        </span>
      </li>
      <li
        class=""
      >
        <span>
          3
        </span>
      </li>
      <li
        class=""
      >
        <span>
          4
        </span>
      </li>
      <li
        class=""
      >
        <span>
          5
        </span>
      </li>
      <li
        class=""
      >
        <span>
          6
        </span>
      </li>
      <li
        class=""
      >
        <span>
          7
        </span>
      </li>
      <li
        class=""
      >
        <span>
          8
        </span>
      </li>
      <li
        class=""
      >
        <span>
          9
        </span>
      </li>
      <li
        class=""
      >
        <span>
          10
        </span>
      </li>
      <li
        class=""
      >
        <span>
          11
        </span>
      </li>
      <li
        class=""
      >
        <span>
          12
        </span>
      </li>
      <li
        class=""
      >
        <span>
          13
        </span>
      </li>
      <li
        class=""
      >
        <span>
          14
        </span>
      </li>
      <li
        class=""
      >
        <span>
          15
        </span>
      </li>
      <li
        class=""
      >
        <span>
          16
        </span>
      </li>
      <li
        class=""
      >
        <span>
          17
        </span>
      </li>
      <li
        class=""
      >
        <span>
          18
        </span>
      </li>
      <li
        class=""
      >
        <span>
          19
        </span>
      </li>
      <li
        class=""
      >
        <span>
          20
        </span>
      </li>
      <li
        class=""
      >
        <span>
          21
        </span>
      </li>
      <li
        class=""
      >
        <span>
          &gt;
        </span>
      </li>
    </ul>
  </nav>
</div>
`;
